x-pack/lib/config_management/elasticsearch_source.rb

```diff
@@ -193,5 +186,63 @@ def client
         @client ||= build_client
       end
     end
+
+    class SystemIndicesFetcher
```

Comment:
I wonder if this might be simpler if we kept the response object in this class, and had methods like 'config_exists?(pipeline_id)', `get_pipeline_config(pipeline_id)` and `get_pipeline_settings(pipeline_id)`. This would avoid having to pass around the `response` and `fetcher` objects

Reply:
This question to me is if we want to further refactor the existing code or the goal is to apply the new API in a manageable way. This involved thirty lines of code, mainly moving `get_pipeline` and the fetcher in a OO way, which is not a big change. At the same time, the readability of the current version is quite similar to the existing one. I am opened to the suggestion. Do you think we should refactor the code?

Reply:
I'm not sure there is a huge amount of refactoring to the existing code either way, beyond what is already present; you already have the method `fetcher.get_single_pipeline_setting(response, pipeline_id)["pipeline"]`
which could change to something like `fetched_config.get_pipeline_settings(pipeline_id)`, although I do realize that there is more work to be done in the extra classes that you have added.

I'm comfortable either way, what you have appears to be functionally correct after running this code locally against Elasticsearch `7.9` and `8.0`